src/components/StatedBalanceForm.tsx

```diff
@@ -1,20 +1,25 @@
 import React from 'react';
+import { Redirect } from 'react-router-dom';
 import BalanceForm from './BalanceForm';
 import BalanceService from './BalanceService';
 
 type Props = { balanceService: BalanceService };
-type State = { value: number };
+type State = { value: number; redirect: string };
 
 class StatedBalanceForm extends React.Component<Props, State> {
   constructor(props: Props) {
     super(props);
-    this.state = { value: 0 };
+    this.state = { value: 0, redirect: '' };
   }
 
   render() {
     const callback = (value: number) => {
       this.props.balanceService.addBalance(value);
+      this.setState({ redirect: '/graph' });
     };
+    if (this.state.redirect) {
+      return <Redirect to={this.state.redirect} />;
+    }
```

Comment:
Implementacja Redirect: Zapraszam do komentowania jako, że jest to jeden ze sposobów na zrobienie tego

Reply:
As for me it is ugly that rendering component gives side effect of redirection.
Also it narrows possibility of executing redirection to React components.

I'd go for something more flexible, like some service giving ability to change urls. 
Then we have a flexibility to choose if we want to change URL in react component or in some other service.

Reply:
But how do you consider mentioned in the article that it is the only recommended way of doing this. At least that this is the most fitting with React methodology

Reply:
I've never met a React component that is having side effects when created.
If something in React documentation states otherwise, then it may be worth reconsidering - however I think not.
So I am not sure if it fits to React methodology.

IMO fitting to React methodology is executing callbacks that mutate state of parent (or parent-parent) component (like a Router). 
Thanks to MobX also we can workaround props-drilling and pass state from anywhere to anywhere.

Therefore passing history (which is a state of Router) via some RouterService to anything that may need it 
(where RouterService usually will be executed from some React component)
is closer to executing callbacks-to-parent-component (which is React methodology),
than creating side effects on component creation is close to any React methodology.

Reply:
So, to specify. Will it be okay if I will create routingService and I will be calling it from level of component instead of Redirect?